newIDE/app/src/ResourcesList/LocalPiskelBridge.js

```diff
@@ -47,17 +52,30 @@ export const openPiskel = ({
   ipcRenderer.removeAllListeners('piskel-changes-saved');
   ipcRenderer.on(
     'piskel-changes-saved',
-    (event, outputResources, newAnimationName) => {
+    (event, outputResources, newAnimationName, metadata) => {
+      const newMetadata = metadata.data
+        ? {
+            pskl: metadata,
+          }
+        : {};
+
       const resourcesManager = project.getResourcesManager();
       outputResources.forEach(resource => {
-        const imageResource = new gd.ImageResource();
         resource.name = path.relative(projectPath, resource.path); // Still needed for onChangesSaved()
-        imageResource.setFile(resource.name);
-        imageResource.setName(resource.name);
-        resourcesManager.addResource(imageResource);
-        imageResource.delete();
+        createOrUpdateResource(project, new gd.ImageResource(), resource.name);
       });
-      onChangesSaved(outputResources, newAnimationName);
+
+      // in case this is for a single frame object, save the metadata in the Image object
+      if (metadata.singleFrame) {
+        resourcesManager
+          .getResource(path.relative(projectPath, outputResources[0].path))
+          .setMetadata(JSON.stringify(newMetadata));
+
+        onChangesSaved(outputResources, newAnimationName);
+      } else {
+        // if not, the metadata will still be passed onto SpritesList.js to be set in the Direction object
```

Comment:
In theory, metadata will be passed to whatever the editor is :) Let's write

// In case there are multiple frames, pass back the metadata to the editor and let it store it at an appropriate place.
// (For example, for sprites, SpritesList.js will save it in the metadata of the `gd.Direction`).